src/arrayMethodSort.js

```diff
@@ -4,8 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let isSwapped;
+
+    do {
+      isSwapped = false;
+
+      for (let i = 0; i < this.length; i++) {
+        const condition = compareFunction
```

Comment:
It's good that you're checking if a 'compareFunction' is provided. However, you should also ensure that the 'compareFunction' is indeed a function before trying to invoke it. You can do this by using 'typeof compareFunction === 'function''.